### PR TITLE
[FIX] web: correct form style selector

### PR DESCRIPTION
**Pull request by Iucapad**
Since o_form_view class is not placed on the same element, depending if the form is displayed in a dialog or not, some styles were not applied in form views. Now, the selector will match all cases and dialogs will be stylized as expected.

### CODE REVIEW

**Comment by Iucapad** [comment]
Before the fix:
image

After the fix: 
image

(The style is applied to .o_form_view.o_form_editable....)

File: addons/web/static/src/views/form/form_controller.scss [hunk 1]
```diff
@@ -130,12 +130,9 @@
     }
 
     .o_row > div > .o_field_widget,
-    &.o_form_editable .oe_title .o_field_char {
-        width: 100%;
-    }
 
     // Readonly specific rules
-    .o_form_readonly {
+    &.o_form_readonly, .o_form_readonly {
```
Comment (mcm-odoo) [inline addons/web/static/src/views/form/form_controller.scss:135]: `.o_form_view.o_form_readonly` is only present in legacy

File: addons/web/static/src/views/form/form_controller.scss [hunk 2]
```diff
@@ -172,13 +169,17 @@
     }
 
     // Editable specific rules
-    .o_form_editable {
+    &.o_form_editable, .o_form_editable {
```
Comment (mcm-odoo) [inline addons/web/static/src/views/form/form_controller.scss:172]: `.o_form_view.o_form_editable` is only present in legacy

**Comment by Iucapad** [comment]
@mcm-odoo Ok, it should be better now. In X2ManyDialogs, the "o_form_view" classname is placed on the dialog container instead of the first div.
The style will now work properly as in the above screenshot

**Comment by Iucapad** [comment]
@mcm-odoo I had to adapt some tests, but it should be fine now. Can you check or approve?

**Comment by mcm-odoo** [comment]
@robodoo r+

**Pull request closed without merging**